src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunctionSet.cs

```diff
@@ -0,0 +1,358 @@
+                {
+                    var fList = WithName(key);
+                    var newFuncs = functionSet.WithName(key);
+
+                    if (fList.Any())
+                    {
+                        fList.AddRange(newFuncs);
+                    }
+                    else
+                    {
+                        _functions.Add(key, newFuncs);
+                    }
+
+                    _count += newFuncs.Count();
+                }
+
+                foreach (var key in functionSet.InvariantKeys)
+                {
+                    var fInvariantList = WithInvariantName(key);
+                    var newFuncs = functionSet.WithInvariantName(key);
+
+                    if (fInvariantList.Any())
+                    {                        
+                        fInvariantList.AddRange(newFuncs);
+                    }
+                    else
+                    {
+                        _functionsInvariant.Add(key, newFuncs);
+                    }
+                }
+
+                foreach (var key in functionSet.NamespaceKeys)
+                {
+                    var fnsList = WithNamespace(key);
+                    var newFuncs = functionSet.WithNamespace(key);
+
+                    if (fnsList.Any())
+                    {
+                        fnsList.AddRange(newFuncs);
+                    }
+                    else
+                    {
+                        _namespaces.Add(key, newFuncs);
+                    }
+                }
+
+                _enums.AddRange(functionSet._enums);
+            }
+
+            return this;
+        }
+
+        internal void Add(IEnumerable<T> functions)
+        {
+            foreach (var t in functions)
+            {
+                Add(t);
+            }
+        }
+
+        internal Dictionary<string, List<T>>.KeyCollection Keys => _functions.Keys;
+
+        internal Dictionary<string, List<T>>.KeyCollection InvariantKeys => _functionsInvariant.Keys;
+
+        internal Dictionary<DPath, List<T>>.KeyCollection NamespaceKeys => _namespaces.Keys;
+
+        internal List<T> WithName(string name) => _functions.ContainsKey(name) ? _functions[name] : new List<T>();
+
+        internal List<T> WithName(string name, DPath ns) => _functions.ContainsKey(name) ? new List<T>(_functions[name].Where(f => f.Namespace == ns)) : new List<T>();
+
+        internal List<T> WithInvariantName(string name) => _functionsInvariant.ContainsKey(name) ? _functionsInvariant[name] : new List<T>();
+
+        internal List<T> WithInvariantName(string name, DPath ns) => _functionsInvariant.ContainsKey(name) ? new List<T>(_functionsInvariant[name].Where(f => f.Namespace == ns)) : new List<T>();
+
+        internal List<T> WithNamespace(DPath ns) => _namespaces.ContainsKey(ns) ? _namespaces[ns] : new List<T>();
+
+        internal List<string> Enums => _enums;
+
+        internal TexlFunctionSet<TexlFunction> ToTexlFunctions()
```

Comment:
>ToTexlFunctions [](http://example.com/codeflow?start=47&length=15)

This is another thing we may be able to remove if we remove the generic parameter